core-bundle/src/Resources/contao/controllers/BackendHelp.php

```diff
@@ -101,9 +101,18 @@ public function run()
 						$rows[] = array('headspan', $arrData['reference'][$key]);
 					}
 
-					foreach ($option as $opt)
+					foreach ($option as $id=>$opt)
 					{
-						$rows[] = $arrData['reference'][$opt] ?? null;
+						if ($key == 'image_sizes')
+						{
+							// Allow using the ID (e.g. "_video_placeholder") as well as the generated
+							// name (e.g. "video_placeholder (1280x)") as key (see #5239)
+							$rows[] = $arrData['reference'][$opt] ?? $arrData['reference'][$id] ?? array($opt, '');
+						}
+						else
+						{
+							$rows[] = $arrData['reference'][$opt] ?? null;
+						}
```

Comment:
```suggestion
						$rows[] = $arrData['reference'][$opt] ?? $arrData['reference'][$id] ?? array($opt, '');
```

Reply:
Except we need `null` as default for non image sizes and we only need `$arrData['reference'][$opt]` as BC layer for image sizes and not in general.

Reply:
Because `$arrData['reference'][$id]` is what we want to use in general.

Reply:
Not sure what you mean. In the default `else` condition, `$arrData['reference'][$id]` would not be used at all?

---

## Here's an example

**References:**
```php
[
    'foo' => ['Foo Title', 'Foo explanation'],
    'bar' => ['Bar Headline', 'Bar explanation'],
]
```

**This `options` array is supported:**
```php
[
    'foo',
    'bar',
]
```

**This `options` does not work right now, but it would be with the changes:**
```php
[
    'foo' => 'Foo label',
    'bar' => 'Bar label',
]
```

Reply:
But this is not how the predefined images array looks like. 🤷‍♂️

Reply:
It does… at least to some extent. Here's an mockup of predefined images (I'm sure you'll see the similarities):

```php
[
	'image_sizes' => [
		'_foo' => 'Foo label',
		'_bar' => 'Bar label',
	],
	'Contao Official Demo' => [
		1 => 'asdf (x)'
	],
	'relative' => [
		0 => 'proportional',
		1 => 'box'
	],
	'exact' => [
		0 => 'crop',
		1 => 'left_top',
		2 => 'center_top',
		3 => 'right_top',
		4 => 'left_center',
		5 => 'center_center',
		6 => 'right_center',
		7 => 'left_bottom',
		8 => 'center_bottom',
		9 => 'right_bottom',
	],
];
```

If you see a problem with the numeric keys, we could also check for `array_is_associative` or `is_number` on the `$id`.

Reply:
When we check for a translation, we only want to check `$arrData['reference']['_foo']` in the future. However, right now, we check for `$arrData['reference']['Foo label']`, which we cannot stop doing until Contao 6 (BC).

Therefore, if the key is `image_sizes`, we have to check

```php
$arrData['reference']['Foo label'] ?? $arrData['reference']['_foo'] ?? ['Foo label', '']
```

but in all other (generic) cases, we only want to check

```php
$arrData['reference']['_foo'] ?? ['Foo label', '']
```

And that is why the `BackendHelp` class must know about `image_sizes` as long as the BC layer remains in place.

Reply:
So in the future the code should look like this?

```php
if (array_is_list($options))
{
	$rows[] = $arrData['reference'][$opt] ?? null;
}
else
{
	$rows[] = $arrData['reference'][$id] ?? array($opt, '');
}
```

As we IMO need the `array_is_list()` check to differentiate between `$arrData['reference']['Foo label']` and `$arrData['reference']['_foo']`

Reply:
Yes, but with the BC layer:

```php
if (array_is_list($options))
{
	$rows[] = $arrData['reference'][$opt] ?? null;
}
else
{
	if ($key == 'image_sizes' && isset($arrData['reference'][$opt]))
	{
		trigger_deprecation('contao/core-bundle', '5.3', 'Using … has been deprecated …');

		$rows[] = $arrData['reference'][$opt];
	}
	else
	{
		$rows[] = $arrData['reference'][$id] ?? array($opt, '');
	}
}
```

Of course that would also work without the `$key == 'image_sizes'` check, I‘m just not sure if there would be side-effects.

Reply:
I’d use your version without the `image_sizes` then (as others may already have used it that way for other options), but only if `$arrData['reference'][$id]` is not set (to make it the preferred one).

```php
if (array_is_list($option))
{
	$rows[] = $arrData['reference'][$opt] ?? null;
}
else
{
	if (isset($arrData['reference'][$opt]) && !isset($arrData['reference'][$id]))
	{
		trigger_deprecation('contao/core-bundle', '5.3', 'Using … has been deprecated …');

		$rows[] = $arrData['reference'][$opt];
	}
	else
	{
		$rows[] = $arrData['reference'][$id] ?? array($opt, '');
	}
}
```

Reply:
I have decided to only fix the issue in Contao 4.13 because there is more that needs to be changed to properly handle image sizes. I will create a follow-up PR for Contao 5.3.